src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -272,7 +272,8 @@ public void deletePullRequestApproval(String pullRequestId) throws IOException,
      */
     @Override
     public boolean checkPathExists(@NonNull String branch, @NonNull String path) throws IOException, InterruptedException {
-        int status = getRequestStatus(V1_API_BASE_URL + owner + "/" + repositoryName + "/raw/" + branch + "/" + path);
+        int status = getRequestStatus(V1_API_BASE_URL + owner + "/" + repositoryName + "/raw/" +
+                branch.replaceAll(" ", "%20") + "/" + path);
```

Comment:
Good point, I couldn't find a list of legal characters with a brief google search (and brief hg source code search), but manually I was able to find that ~ ` ! @ # $ % ^ & * ( ) + = [ ] { } \ | ; " < > , . / \ ? are all legal characters that need to be encoded. Only illegal character I've found is ':'

Repeated test with a branch name '~`!@#$%^&*()_+=[]{}\|;"<>,./\?a' and branch scanning fails with the same error.